### ISSUE TITLE
Some tag issues

### ISSUE THREAD

**Issue by davidfarmer** [rbeezer/mathbook#711]
This page
   http://physics.thomasmore.edu/ConnectedPhysics/ss-NI.html
has some HTML markup issues.

1.  The second assemblage "Forces are those things that cause a change in the velocity."
has an empty title.  This messes with the spacing.

2.  The words immediately after it are not in a "p", but they should be.

3.  After that collection of words is an empty "p"

I don't know what the PTX source looks like, and there is no way for me to tag the
author as I submit this issue.

**Comment by davidfarmer** [comment]
Joe: If you built the HTML with old PTX, please update and re-build, since some of these
items may have already been fixed.

**Comment by rbeezer** [comment]
@sepvar

**Comment by sepvar** [comment]
Hi, @davidfarmer 
Thanks for the comments.  For context, I spent two months this summer writing the book in LaTeX before learning about and translating the document into XML, with all of the attendant Windows installation issues (on three separate windows machines!)... So this weekdn is the first time I have been able to install and run jing.  I have 700 errors to work through even though the xsltproc compiles without complaint.  Yes, @rbeezer I am working to verify my code.  I'll get there; thank you for your patience.

To @davidfarmer 's comments: 
1. that was (originally) intentional.  I only wanted a colored box to act as a highlighter on important statements that were not SO important as to warrant an actual <note> or <property> or <definition>; I was just trying to catch the reader's eye if they were skimming, instead of reading.  I know I need to change it, but I am not sure how I want to do that yet.

This is the current raw code:

<p xml:id="d-atrestinmotion">The rest of this statement is often written a little differently (and less concisely) as <q>an object at rest remains at rest unless acted on by an external force and an object in motion remains in motion unless acted on by an external force.</q>  Since being <xref ref="d-atrest" text="title"><q>at rest</q></xref> is a statement about the velocity<m>(\vec v=0)</m> and being <xref ref="d-motion"><q>in motion</q></xref> is also a statement about the velocity, each of these statements can be understood as saying that
<assemblage><p>Forces are those things that cause a change in the velocity.</p></assemblage>
In other words, Newton's first law says that without a force, the velocity will not change.  In the discussion of <xref ref="sss-equilibrium" text="title">equilibrium</xref>, we will note that this is often extended to say that without a <xref ref="sss-netforce" text="title">net force</xref> the velocity will not change, but that is a special case of <xref ref="ss-NII" text="title">Newton's second law</xref>.</p>

I know I can't xml:id a <p> and I will change it to a <paragraphs>
I know I need to </p> before and <p> after the <assemblage>
I'm working on it.

I think **the point of this page** at this point is only to show the way I use the <aside><title>Touchstone</title><p>words</p></aside>

Unless you have a different comment, I think you can close this issue?

**Comment by davidfarmer** [comment]
Joe: I can't actually read what you wrote, because the tags are making things invisible.
How to we make issue contents "raw" in GitHub?

The first point of this issue is HTML markup problems.  Can you update PTX, and post
updated HTML?  Otherwise we can't see where the problem is.

I might have more to say if only I could read your previous comment :)

**Comment by rbeezer** [comment]
"Styling With Markdown" below comment box is a link to a simple guide.

Three backticks, languge (ie "xml"), then newline.
Raw code.
Newline, three backticks.

**Comment by sepvar** [comment]
Whoops!  Thanks.

```<p xml:id="d-atrestinmotion">The rest of this statement is often written a little differently (and less concisely) as <q>an object at rest remains at rest unless acted on by an external force and an object in motion remains in motion unless acted on by an external force.</q>  Since being <xref ref="d-atrest" text="title"><q>at rest</q></xref> is a statement about the velocity<m>(\vec v=0)</m> and being <xref ref="d-motion"><q>in motion</q></xref> is also a statement about the velocity, each of these statements can be understood as saying that
<assemblage><p>Forces are those things that cause a change in the velocity.</p></assemblage>
In other words, Newton's first law says that without a force, the velocity will not change.  In the discussion of <xref ref="sss-equilibrium" text="title">equilibrium</xref>, we will note that this is often extended to say that without a <xref ref="sss-netforce" text="title">net force</xref> the velocity will not change, but that is a special case of <xref ref="ss-NII" text="title">Newton's second law</xref>.</p>```

**Comment by sepvar** [comment]
Whoops.  Thanks.

```
<p xml:id="d-atrestinmotion">The rest of this statement is often written a little differently (and less concisely) as <q>an object at rest remains at rest unless acted on by an external force and an object in motion remains in motion unless acted on by an external force.</q>  Since being <xref ref="d-atrest" text="title"><q>at rest</q></xref> is a statement about the velocity<m>(\vec v=0)</m> and being <xref ref="d-motion"><q>in motion</q></xref> is also a statement about the velocity, each of these statements can be understood as saying that
<assemblage><p>Forces are those things that cause a change in the velocity.</p></assemblage>
In other words, Newton's first law says that without a force, the velocity will not change.  In the discussion of <xref ref="sss-equilibrium" text="title">equilibrium</xref>, we will note that this is often extended to say that without a <xref ref="sss-netforce" text="title">net force</xref> the velocity will not change, but that is a special case of <xref ref="ss-NII" text="title">Newton's second law</xref>.</p>
```

**Comment by sepvar** [comment]
Whoops.  Thanks.  I hope you are not getting alerted every time I try and delete versions of this.

Revision edited above

**Comment by Alex-Jordan** [comment]
Regarding posts with tags that accidentally get eaten. On other repos where I am an owner, I get a pencil icon with each post indicating I can edit it. If I suspect someone used tags because what I see seems to have holes, I can edit their comment, see it as they typed it, and add backticks. Since David deals with tags often in these conversations, maybe Rob could look into the possibilities for granting David adequate permission to edit comments?

**Comment by sepvar** [comment]
Howdy.

So, I fixed the issue above by changing the way I set it up; but I am encountering the issue again.  I created a minimal example:  http://physics.thomasmore.edu/ConnectedPhysics/minimal/minimal.html

So, I want to link back to two literal-paragraphs, one is a "p" before an "investigation" and the other is a "p" after the "investigation", so I wrapped the "p"-"investigation"-"p" in a "paragraphs" and linked to that.  Yay, solved.  Except that this group happens to be in an introduction, rather than in a section.  "paragraphs" are allowed in sections/subsections, etc, but are not allowed in "introduction"s.  So, I solved that by putting the "paragraphs" in a "sidebyside", but the sidebyside makes the "investigation" disappear in the text.  I think investigations are allowed in the paragraphs inside the sidebyside and it does appear in the knowl, just not in the text.

Thoughts?

Thanks

Joe

**Comment by davidfarmer** [comment]
My first reaction is that maybe it is good to prohibit paragraphs in
an introduction, because maybe there is some other structural element
which is common in introductions.  Figuring out what that is,
and then perhaps renaming it, might be a good option.

If paragraphs are not allowed in introduction, but sidebyside is, and that
allows paragraphs, then either we allow paragraphs (which, as I said
above, I have not yet seen the justification), or else we eliminate
that "sidebyside trick".

**Comment by sepvar** [comment]
**This is my argument for allowing paragraphs**:  The reason I am trying to write an online book is that I specifically want to be able to call back and look ahead so that students can see the interdependence between the material.  I am specifically intending to call back and look ahead to topics that are multiple chapters away from the current conversation.

With that, I am finding that I like being able to call back (like the "touchstone" idea) to a knowl with sometimes two paragraphs or, like in this case, to a paragraph that is tied to an adjacent environment.  I don't want the student to have to actually visit the section because that could be disruptive, but seeing the knowl I hope to spark the memory of the previous discussion in a way that a single paragraph might not capture if the call-back is from an earlier chapter, especially if it is fairly distant (such as happens in a two-semester course).  

Since my introduction is a reasonable place to touch intuition or to lay out big-picture concepts, I would like the call-back to not be restricted to subsections.  

I think what I am saying is that if we allow "paragraphs" in sections and subsections, I don't understand why we wouldn't allow them in "introduction"s.  How do you see an "introduction" as so fundamentally different from a "subsection"?  To my mind the "introduction" and "conclusion" of a section are equivalent to unnamed "subsection"s.  The allowable elements of a section that are not allowable in an introduction (or, by the way, a conclusion) are:
conclusion, introduction, exercises, objectives, paragraphs, references, subsection
I understand all of those except paragraphs.

Maybe I can say it a different way.  What is the point of the paragraphs element if not to group "p"s and other elements together and why wouldn't we want that in an introduction (or conclusion)?  And if we want to group them, then it seems that we ought to be able to reference them.

**This is my comment about sidebyside**: I would be perfectly happy wrapping my paragraphs in a sidebyside, which I assume others like having in the introduction/conclusion.  The problem with this "trick" is that it seems to swallow the investigation that I included...

**Comment by davidfarmer** [comment]
I don't think my previous point came across.

If most of the paragraphs you want to put in an introduction are
"key concept"s, then that suggests there should be a special type of
environment in the introduction.  The "key concepts" environment
would be special, in the same way that "learning goals" (I forget
if that is the official name) are a special type of environment,
designed to go at the beginning of a section.

So my request to you is to not try to fit what you want to do
into the existing framework.  Rather, think about what *structural*
elements are not accurately captured by the PXT markup, and then
let's discuss how to handle those objects.

If the structure is marked up well, then you will be able to
do what you want.  But get the source marked up properly first,
and then worry about how to do what you want.

As to the PXT use of "introduction" and "conclusion": those are
to address a defect in LaTeX.  An "introduction" exists implicitly
in LaTeX: it is the material before the first (sub)section.
But "conclusion" does not exist in LaTeX: that language has no
way to mark up material that is after the last subsection
but before the end of the section.  (I am just trying to get across
the point that thinking in terms of LaTeX is harmful.  We need to
think in terms of the abstract structure of documents, and then try
to capture what we can in PTX.)

**Comment by sepvar** [comment]
Ah.  I see.  (For completeness, your reference to "learning goals" is called "objectives".) I did miss that in my original reading and I am glad you clarified it.  At the same time, I don't think that what I am trying to accomplish could be done in LaTeX, so I don't think I am thinking in those terms.

I do not have a name for the element I am looking for (equivalent of "objectives"), and it will not always be the same, such as "key concepts".  It is, however, always a paragraph or two.  When it is a single "p", that's easy.  Sometimes it is the pairing of two ideas, ie two "p"s, which we have a name for: "paragraphs" and which I can use in sections and subsections, but not in introductions.  I would prefer to not require a title for "paragraphs" and I would like to use that moniker for "this group of 'p' that I would like to reference later".

I am reading through my book looking for other possible examples.  I found two possible examples (listed below although the specifics are not particularly relevant).  In both of these cases, the thing that I would reference is more than just a "p".  In both of these cases, I can wrap them in a "paragraphs" to get the effect I want without having to define a specific structure (which I don't see as necessary).  In both of these cases, "paragraphs" works only because I do not happen to have "subsubsections" beneath these "subsections".  If I did want to supplement the conversation with "subsubsections" then these words would necessarily be in an "introduction", which does not allow a "paragraphs" and would necessitate creating an environment that is otherwise not necessary.

**Example 1** On page http://physics.thomasmore.edu/ConnectedPhysics/sss-inertia.html, the "Insight" itself is the main point that I want students to walk away with when they read this subsection, but when we get to the chapter on momentum, calling up only the insight will likely be insufficient and I will probably want to call up the previous "p" as well as the "insight" statement.  At the same time, I don't think that the "p" rises to the level of calling it a "clarification".  (Although that particular example could be solved by referencing the full section since it is so short.)

**Example 2** On page http://physics.thomasmore.edu/ConnectedPhysics/ss-weightmass.html, there are two clarifications (8.1.3 and 8.1.4) which grammatically are part of a single paragraph, but technically are not within the "p" and would need to be within a "paragraphs" in order to connect them with the adjacent "p"s.

**Comment by davidfarmer** [comment]
I think I understand now.  There are several things going on.
Here is a summary, and I will start a thread on pretext-dev.

In your example 1, the paragraph before the Insight can be the "preamble"
to the Insight.  That attaches it to the Insight and so it can appear
in the knowl.  (Maybe that doesn't work yet, but it will, because that
is the right thing to do.)

In your example 2, independently of the markup issue, I suggest you
rewrite it so that each Clarification contains a full sentence.  I also
suggest "term" tags around 'mass' and 'weight' in each clarification.
Also consider putting both sentences into one Clarification.  Those
edits will make it easier on the reader.
If you rewrite it, then preamble and postamble can attach the surrounding
paragraphs.

More generally, what I think you need is a "keyidea" tag that serves
the purpose you are using for most of your "paragraphs".  For example,
the knowl after the second Clarification, 'fundamental unit', would be
a keyidea, not a paragraphs.

**Comment by rbeezer** [comment]
A response to a question in this thread.  `introduction` and `paragraphs` are both divisions and they are both terminal.  `introduction` is meant to be simple - a *very brief* description of what is coming.  You can put a fair amount of stuff into a `paragraphs`.  So you cannot divide an `introduction` with a `paragraphs`.

And `paragraphs` in a `sidebyside` will be going away, as a consequence of requests Joe C made a while ago.

You can always
* Not use an `introduction` at all
* Title a first division "Introduction"
* Load that division with lots of complex content

**Comment by sepvar** [comment]
Hi Rob.
I have grown to understand your intent with introductions and with paragraphs.  (Thanks to you and David for your patience with my "enthusiastic variations"...)

I think that many of my difficulties related to this thread and others would be resolved with the `prelude` and `postlude` added to most complex types (TheoremLike, AxiomLike, etc, although I don't see a need in AsideLike).  I think https://github.com/rbeezer/mathbook/issues/720 houses that discussion.

I am also learning better ways to reference paragraphs (both "p" and "paragraphs") in that if it is worth referencing, it is probably worth putting into a `remark` or some such environment.

It may be that this issue can be closed in favor of #720?
Joe

**Comment by rbeezer** [comment]
House-cleaning:
*  #720 could use broader implementation.
*  `paragraphs` in `sidebyside` are now banned, see 524e9591d28bfc4decde53fe58be868cc8647082 and 9eb7ca39d9011f06c1c917410873e22787403762, in favor of `p` in `stack`
*  Unsure if we are happy with current state of `xml:id` on `p`, but if not, that conversation should restart somewhere else.

So I think this one is done.
